src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Math.round(Double.valueOf(s)) : Long.valueOf(s);
```

Comment:
Rather than rounding it would be more consistent to use Double.longValue, since in 0.6.x we always cast to long before aggregating in longSum.